### PR TITLE
feat: intergrate scanning for v1 and v2 APIs

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx [hunk 1]
```diff
@@ -32,19 +32,15 @@ const Vulnerabilities = ({
...
-            artifactId: String(artifactId),
-            envId: environmentId,
-            isScanEnabled,
-            isScanned,
-            isScanV2Enabled,
-            getSecurityScan,
-        })
+    const [scanResultLoading, scanResultResponse, scanResultError, reloadScanResult] = useAsync(
+        () => getSecurityScan({ artifactId, appId: applicationId }),
+        [],
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx:38]: Do we need to add dep?

File: src/Shared/Components/Security/SecurityModal/service.ts [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-            })) || [],
-    }
-    return { ...response, result: parseExecutionDetailResponse(parsedResult) }
+export const getSecurityScan = async ({
+    appId,
+    envId,
+    installedAppId,
+    artifactId,
+    installedAppVersionHistoryId,
+}: AppDetailsPayload): Promise<ResponseType<ScanResultDTO>> => {
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/SecurityModal/service.ts:13]: Can add type to payload

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx (hunk 1 above)
Comment (arunjaindev) [reply src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx:38]: not req, in a map call will be different for each material

File: src/Shared/Components/Security/utils.tsx [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
-})
+import { SCAN_TOOL_ID_CLAIR, SCAN_TOOL_ID_TRIVY } from '@Shared/constants'
+import { ScanResultDTO, SeveritiesDTO } from './SecurityModal/types'
 
 export const getCVEUrlFromCVEName = (cveName: string): string =>
     `https://cve.mitre.org/cgi-bin/cvename.cgi?name=${cveName}`
+
+export const getScanToolAndSeverityCount = (
+    scanResult: ScanResultDTO,
+): { scanToolId: number; severityCount: SeverityCount; totalCount: number } => {
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/utils.tsx:10]: Suggestion: Can we create a return type interface?

File: src/Shared/Components/Security/utils.tsx [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
+    const scanToolId =
+        scanResult?.imageScan?.vulnerability?.list?.[0].scanToolName === 'TRIVY'
```
Comment (AbhishekA1509) [inline src/Shared/Components/Security/utils.tsx:12]: Suggestion: Can create a variable: scanResult?.imageScan?.vulnerability

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**